FabNESO/ensemble_tools.py

```diff
@@ -84,28 +97,58 @@ def create_dict_sweep(
     if destructive and sweep_path.is_dir():
         shutil.rmtree(sweep_path)
     # Uniformly spaced grids on [low, high] for each parameter
-    parameter_grids = (
-        {key: [low] for key, (low, high) in parameter_dict.items()}
-        if n_divs == 1
-        else {
-            key: [low + (i / (n_divs - 1)) * (high - low) for i in range(n_divs)]
-            for key, (low, high) in parameter_dict.items()
-        }
-    )
+    parameter_grids = {
+        key: [calculate_parameter_value(n_dirs, low, high, i) for i in range(n_dirs)]
+        for key, (low, high) in parameter_dict.items()
+    }
     # Compute Cartesian products of all parameter value combinations plus grid indices
     for parameter_values, indices in zip(
         _product_dict(parameter_grids),
-        itertools.product(*(range(n_divs),) * len(parameter_dict)),
+        indices_iterator(n_dirs, len(parameter_grids)),
         strict=True,
     ):
-        directory_name = "-".join(
-            f"{k}_{i}" for k, i in zip(parameter_values, indices, strict=True)
-        )
+        directory_name = return_directory_name(parameter_values, indices)
         directory_path = Path(sweep_path) / "SWEEP" / directory_name
         shutil.copytree(copy_dir, directory_path)
         edit_parameters(directory_path / edit_file, parameter_values)
 
 
+def return_directory_name(parameter_values: dict, indices: tuple[Any, ...]) -> str:
+    """Return the directory name given parameter names and indices."""
+    return "-".join(f"{k}_{i}" for k, i in zip(parameter_values, indices, strict=True))
```

Comment:
As this function only requires the keys (parameter names) from `parameter_values` dictionary, I think it would be more readable and also simplify testing to just accept an iterable of strings as an argument `parameter_names`, and when calling just pass `parameter_values.keys()`.

Reply:
This shouldn't require any changes to implementation other than changing parameter name and updating type hint.

Reply:
I have had to cast the `.keys()` as a list to satiate `mypy`, but otherwise this has been done.

Reply:
That works - in case it's helpful in future `Iterable` in `collections.abc` should I think be a valid type hint for either a list or `dict_keys` (or any object which implements an `__iter__` method).